client/src/components/RelatedProducts/ProductCard.jsx

```diff
@@ -4,72 +4,156 @@ import { makeStyles } from '@material-ui/core/styles';
 import Card from '@material-ui/core/Card';
 import CardMedia from '@material-ui/core/CardMedia';
 import CardContent from '@material-ui/core/CardContent';
+import Rating from '@material-ui/lab/Rating';
+import AverageRating from '../AverageRating.jsx';
+import Icon from '@material-ui/core/Icon';
+import Modal from '@material-ui/core/Modal';
+import ComparisonModal from './ComparisonModal.jsx';
+
 
 const useStyles = makeStyles((theme) => ({
   root: {
-    maxWidth: 230,
-    maxHeight: 350,
+    width: 230,
+    height: 350,
   },
   media: {
     paddingTop: '70.25%',
   },
   content: {
     paddingTop: '10%',
+  },
+  icon: {
+    position: 'relative to parent',
+    top: 5,
   }
 }));
 
 const ProductCard = (props) => {
   const [productCardInfo, setProductCardInfo] = useState(null);
   const [productCardPhoto, setProductCardPhoto] = useState(null);
+  const [ratingsInfo, setRatingsInfo] = useState({
+          avgProductRating: 0,
+          totalRatings: 0
+          });
+  const [currentProductInfo, setCurrentProductInfo] = useState(null);
+  const [showComparison, setShowComparison] = useState(false);
+  const [featureData, setFeatureData] = useState(null);
   const classes = useStyles();
 
+
   const getPhoto = () => {
     axios.get(`/api/products/${props.productId}/styles`)
-      .then(stylesInfo => {
-        setProductCardPhoto(stylesInfo.data.results[0]);
-      })
-      .catch(err => {
-        console.info('There was an error getting product photo from the server.');
-      });
+    .then(stylesInfo => {
+      setProductCardPhoto(stylesInfo.data.results[0]);
+    })
+    .catch(err => {
+      console.info('There was an error getting product photo from the server.');
+    });
   }
 
   const getProductInfo = () => {
     axios.get(`/api/products/${props.productId}`)
-      .then(productInfo => {
-        setProductCardInfo(productInfo.data);
-      })
-      .catch(err => {
-        console.info('There was an error retrieving product information from the server.');
-      });
+    .then(productInfo => {
+      setProductCardInfo(productInfo.data);
+    })
+    .catch(err => {
+      console.info('There was an error retrieving product information from the server.');
+    });
+  }
+
+  const getCurrentProductInfo = () => {
+    axios.get(`/api/products/${props.currentProduct.id}`)
+    .then(productInfo => {
+      setCurrentProductInfo(productInfo.data);
+    })
+    .catch(err => {
+      console.info('There was an error retrieving product information from the server.');
+    });
+  }
+
+  const getFeatureData = () => {
+    const featureData = {};
+
+    for (var i = 0; i < currentProductInfo.features.length; i++) {
+      var currentFeatureObject = currentProductInfo.features[i];
```

Comment:
I personally don't like the mix of es5 with es6 (using var when everywhere else is using const or let) but not a big deal

Reply:
I think I could probably get away with let. I am using the same variable name for different circumstances, so I thought const would be problematic.